### ISSUE TITLE
Inconsistent symbol in scope behaviour among different kinds of fields

### ISSUE THREAD

**Issue by pubudu91** [ballerina-platform/ballerina-lang#29382]
Consider the following code snippet:
```ballerina
public class Foo {
    bar:Unknown c = new();
    f<cursor1>
}

type Person record {|
    bar:Name name;
    f<cursor2>
|};

type PersonObj object {
    bar:Name name;
    f<cursor3>
};
```

If we check the symbols visible at each of the above illustrated points, both first and second cases return 4 symbols (including the `c` and `name` fields). But in cursor3 position, only 3 symbols are returned (just the type defs). What should be the correct behaviour in these 3 cases?

**Comment by MaryamZi** [comment]
Since the cursors are in the context of a field, we shouldn't be seeing the names of the fields, right?

**Comment by pubudu91** [comment]
> Since the cursors are in the context of a field, we shouldn't be seeing the names of the fields, right?

Yeah, I think that makes more sense seeing as we can't use the other fields in those contexts.

**Comment by sanjana** [comment]
Blocked by #33089
